Fluent Calendar

amwx opened the pull request:
## What does the pull request do?
Adds Fluent/WinUI themed Calendar. MS handles some of the styling for calendar day items in code so some styles are based on observed behavior. I didn't know how to test it with the Avalonia source, but it worked in my test project

![image](https://user-images.githubusercontent.com/40413319/83246329-944d0380-a167-11ea-8964-b9bcfb044484.png)



## What is the current behavior?


## What is the updated/expected behavior with this PR?


## How was the solution implemented (if it's not obvious)?


## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

## Breaking changes

## Fixed issues

amwx committed:
Merge pull request #1 from AvaloniaUI/master

Update from original

amwx committed:
Fluent Calendar style

amwx committed:
Fluent Calendar style

amwx committed:
Fluent Calendar Style

amwx committed:
Fluent Calendar Style

dnfclas commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=4028) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=4028) before we can accept your contribution.<br/><br/>:x: amwx [sign now](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=4028)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/AvaloniaUI/Avalonia?pullRequest=4028) it.</sub>

jmacato commented:
It looks like this here
![image](https://user-images.githubusercontent.com/16554748/83253825-37416400-a1e0-11ea-8d22-71171cf34b29.png)

danwalmsley commented:
Doesn't look right, probably missing resources , I'll take a look.

Commit:
add some missing resources for calendar

amwx committed:
Update

amwx committed:
Fix Calendar buttons

amwx commented:
I've updated the Button's to not have rounded corners and center the day numbers & fixed a couple color resources that were incorrect. I've also added the disabled style which I forgot. 
The transitions could be added, but there's a couple issues
1-Calendar doesn't currently have pseudoclasses for each display mode
2-WinUI has 3 separate containers, one for each display mode, Avalonia only has two (year/decade share), so currently transition effects could only work when going month -> year or year -> month
3-Another container could be added, but it would require a change in the default theme style most likely
I'm hesitant to touch the source code cause I don't want to break anything inadvertently as the Calendar is a complex beast

danwalmsley commented:
> I've updated the Button's to not have rounded corners and center the day numbers & fixed a couple color resources that were incorrect. I've also added the disabled style which I forgot.
> The transitions could be added, but there's a couple issues
> 1-Calendar doesn't currently have pseudoclasses for each display mode
> 2-WinUI has 3 separate containers, one for each display mode, Avalonia only has two (year/decade share), so currently transition effects could only work when going month -> year or year -> month
> 3-Another container could be added, but it would require a change in the default theme style most likely
> I'm hesitant to touch the source code cause I don't want to break anything inadvertently as the Calendar is a complex beast

We dont need to update the default theme, as long as we dont break existing functionality...

default theme will be deprecated.

Commit:
Merge branch 'master' into master

Commit:
Merge branch 'master' into master

danwalmsley reviewed (approved):
LGTM, we will have to revisit to add the transitions.

grokys commented on src/Avalonia.Themes.Fluent/CalendarButton.xaml:
```diff
@@ -7,75 +7,115 @@
+            <Setter Property="IsVisible" Value="True"/>
+      </Style>-->
+
+  <Style Selector="CalendarDayButton:disabled /template/ ContentControl#Content">
```
Is this a mistake? Should it be `CalendarButton`?
